### PR TITLE
Get correct version of pulp-cli tests

### LINKED ISSUE #603: OCI images cli tests keep failing.
Instead, I suggest: Install the cli, retrieve it's version and checkout the corresponding release tag to grant access to the matching test suite.

_Originally posted by @mdellweg in https://github.com/pulp/pulp-oci-images/pull/587#discussion_r1477035103_

Currently we run the wrong test suite against the installed version of the CLI, incidentally failing for that reason.

### PR DESCRIPTION
**Pull request by markafarrell**
Fixes #603

### CODE REVIEW

**Comment by markafarrell** [comment]
We have an issue as CI is attempting to use `pulp/pulp-web:3.48.0` but this does not exist on dockerhub.

Not sure how we resolve this.

**Comment by markafarrell** [comment]
> We have an issue as CI is attempting to use `pulp/pulp-web:3.48.0` but this does not exist on dockerhub.
> 
> Not sure how we resolve this.

Seems to have resolved itself. 

😃

File: images/s6_assets/pulp_tests.sh [hunk 1]
```diff
@@ -12,16 +12,25 @@ echo "Base Address: $BASE_ADDR"
...
+
+# Retreive installed pulp-cli version
+PULP_CLI_VERSION=$(python3 -c \
+  'import importlib.metadata; \
+   from packaging.version import Version; \
+   print(Version(importlib.metadata.version("pulp-cli")))')
+
+# Checkout git repo for pulp-cli at correct version to fetch tests
+if [ ! -d pulp-cli ]; then
   git clone --depth=1 https://github.com/pulp/pulp-cli.git
```
Comment (mdellweg) [inline images/s6_assets/pulp_tests.sh]:
```suggestion
git clone --depth=1 https://github.com/pulp/pulp-cli.git -b "${PULP_CLI_VERSION}"
```
You should be able to clone that tag directly.

Comment (markafarrell) [reply images/s6_assets/pulp_tests.sh]: Thanks. I have made the change

**Review by mdellweg — APPROVED** [review]

**Pull request merged**